### DISCUSSION TITLE
[QUESTION] how to use vp8enc?

### DISCUSSION THREAD

**Discussion by mattfeury** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1692] [Q&A] [answered]
Hello,

I am attempting to use the gstreamer sample. It is working well with the default x264enc. However, I'm interested in trying vp8enc. I'm using the [pipeline listed in the comments](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/b820e2ea3d28c670da2fbc789face742cb949ef1/samples/kvsWebRTCClientMasterGstreamerSample.c#L141), but when running I get:

```
2023-04-11 15:44:22 DEBUG   getNextNaluLength(): Warning: Failed to get the next NALu in H264 payload with 0x5c000003
2023-04-11 15:44:22 ERROR   writeFrame(): operation returned status code: 0x5c000003
```

over and over again. seems like writeFrame is expecting a h264 frame, which i'm obviously not using anymore. are there other changes required to get vp8 encoding working?

**Comment by disa6302** [comment, marked as answer]
You also need to set up the appropriate transceivers:

https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/v1.7.3/samples/Common.c#L497. Have you done this?

**Reply by mattfeury** [reply]
i haven't changed the sample code besides the gstreamer pipeline. is it basically just changing `videoTrack.codec`? searched the project but didn't see any documentation on it

**Reply by mattfeury** [reply]
that worked! thank you!!
